### PR TITLE
[WIP] Add presubmit for publishing-bot dry run

### PR DESCRIPTION
**Pull request by nikhita**
This should help us in catching things that break the publishing-bot early.
/cc @sttts @dims 
/hold

### CODE REVIEW

**Comment by nikhita** [comment]
I'm always bitten by https://github.com/kubernetes/test-infra/issues/9469. Should I be concerned about it here as well?

File: config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml [hunk 1]
```diff
@@ -31,3 +32,120 @@ presubmits:
...
+        containers:
+          - image: gcr.io/k8s-testimages/kubekins-e2e:v20190301-76bc03340-master
+            command:
+            - make
+            args:
+            - run
+            env:
+            - name: CONFIG
+              value: "https://raw.githubusercontent.com/kubernetes/publishing-bot/master/configs/kubernetes"
+            - name: DRYRUN
```
Comment (nikhita) [inline config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml]: I'm still at a loss of where this is really being used in the publishing-bot (grepping gives me only the README)....but this is being used in production, so I've added it here. Doesn't hurt anyway.

File: config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml [hunk 2]
```diff
@@ -31,3 +32,120 @@ presubmits:
...
+          repo: sample-controller
+          base_ref: master
+          path_alias: k8s.io/sample-controller
+      spec:
+        containers:
+          - image: gcr.io/k8s-testimages/kubekins-e2e:v20190301-76bc03340-master
+            command:
+            - make
+            args:
+            - run
```
Comment (nikhita) [inline config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml]: @sttts @dims while using `make run`, I had to create the namespace `k8s-publishing-bot` for the script to work. What was I missing?

File: config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml [hunk 3]
```diff
@@ -31,3 +32,120 @@ presubmits:
...
               - /home/prow/go/src/k8s.io/kubernetes/staging/publishing/rules.yaml
+
+  kubernetes/kubernetes:
+    - name: pull-publishing-bot-dry-run
+      always_run: false
+      decorate: true
+      path_alias: k8s.io/kubernetes
+      skip_report: false
+      optional: true
+      run_if_changed: '^staging/publishing.*$'
```
Comment (dims) [inline config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml]: lets please add `run_if_changed` once we get the job working. is that ok?

File: config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml [hunk 4]
```diff
@@ -31,3 +32,120 @@ presubmits:
...
+      extra_refs:
+        - org: kubernetes
```
Comment (dims) [inline config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml]: Here's a problem, what happens when we need to add a new staging repo? will this job break during the process? (and hence not allow the PR with new staging repo to merge?)

could we pull the other repos using a script?

File: config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml (hunk 2 above)
Comment (dims) [reply config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml]: yes, that's true. i had to create the namespace by hand as well

File: config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml (hunk 4 above)
Comment (dims) [reply config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml]: Also we need the job to have way more memory to run (and it needs a volume too ... not sure how that will work)

File: config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml [hunk 5]
```diff
@@ -31,3 +32,120 @@ presubmits:
...
+
+  kubernetes/kubernetes:
+    - name: pull-publishing-bot-dry-run
+      always_run: false
+      decorate: true
+      path_alias: k8s.io/kubernetes
+      skip_report: false
+      optional: true
+      run_if_changed: '^staging/publishing.*$'
+      extra_refs:
```
Comment (stevekuznetsov) [inline config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml]: wowza

File: config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml (hunk 2 above)
Comment (stevekuznetsov) [reply config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml]: Are you expecting to create namespaces in the test cluster?

Comment (nikhita) [reply config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml]: > Are you expecting to create namespaces in the test cluster?

(my bad, should have added a comment about this on Friday)

No...given the other things that this cluster will need (ref @dims' comment here https://github.com/kubernetes/test-infra/pull/11697#discussion_r263776911), can we have another cluster for this job - just like we use `test-infra-trusted` for some jobs? Can wg-k8s-infra help here?

Comment (nikhita) [reply config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml]: @dims [mentioned on slack](https://kubernetes.slack.com/archives/CCK68P2Q2/p1552315195087700) that we could also run the binary directly...I'll look into this after fixing a flake with the publishing-bot. Please don't review yet :see_no_evil:

**Comment by nikhita** [comment]
Will reopen when ready.

**Pull request closed without merging**